packages/lesswrong/components/tagging/SubforumCommentsThread.tsx

```diff
@@ -14,6 +16,19 @@ const SubforumCommentsThread = ({ tag, terms }: {
     fetchPolicy: 'cache-and-network',
     enableTotal: true,
   });
+
+  const currentUser = useCurrentUser();
+  const [recordSubforumViewMutation] = useMutation(gql`
+    mutation recordSubforumView($userId: String!, $tagId: String!) {
+      recordSubforumView(userId: $userId, tagId: $tagId)
+    }
+  `);
+  const recordSubforumView = useCallback(async () => recordSubforumViewMutation({variables: {userId: currentUser?._id, tagId: tag._id}}), [currentUser?._id, tag, recordSubforumViewMutation]);
+
+  useEffect(() => {
+    if (results && results.length)
+      void recordSubforumView();
+  }, [results, recordSubforumView]);
```

Comment:
Is there a reason you are doing this as a separate callback? It seems more efficient to just record it as part of the comments fetch

Reply:
I did initially try making this a side effect (on the server side) of fetching the 'tagSubforumComments' view. This worked but it caused a weird circular dependency issue that I couldn't fix, which was basically a result of trying to do db access from within the `addView` callback. A way of doing this that would work would be to use a feed resolver as a wrapper around the view, but this seemed like more hassle than just adding a separate mutation

On the client side it could also be written in such a way that this mutation is triggered whenever the query is fetched to avoid the separate `useEffect`. But the useQuery/useMulti behaviour makes it not obvious where this actually happens (sometimes it is triggered via `refetch`, sometimes not), so I thought this way was about the simplest way to do it